File: 01b/python/main.py
from itertools import islice


with open('input.txt', 'r') as f:
    data = f.readlines()
nums = [int(d.strip()) for d in data]

sums = [sum(tup) for tup in zip(nums, nums[1:], nums[2:])]
pairs = zip(sums, sums[1:])
result = sum(b > a for (a, b) in pairs)

print(result)

File: 01b/ruby/main.rb
FILENAME = "input.txt"
file = File.open(FILENAME)
lines = file.readlines.map(&:chomp)

increases = 0
last_sum = -1

lines.each_cons(3) do |a|
  sum = a.map(&:to_i).sum
  if last_sum != -1 and sum > last_sum
      increases += 1
  end
  last_sum = sum
end

puts increases
